src/NServiceBus.Core/Routing/MessageDrivenSubscriptions/MessageDrivenSubscriptions.cs

```diff
@@ -54,11 +54,20 @@ protected internal override void Setup(FeatureConfigurationContext context)
             {
                 var distributorAddress = context.Settings.GetOrDefault<string>("LegacyDistributor.Address");
 
-                var subscriberAddress = distributorAddress ?? context.Settings.LocalAddress();
+                var subscribeWithInstanceSpecificQueue = context.Settings.GetOrDefault<bool>("SubscribeWithInstanceSpecificQueue");
+                var instanceSpecificQueue = subscribeWithInstanceSpecificQueue ? context.Settings.InstanceSpecificQueue() : null;
+
+                if (subscribeWithInstanceSpecificQueue && instanceSpecificQueue == null)
+                {
+                    throw new Exception("SubscribeWithInstanceSpecificQueue requires a uniquely addressable instance. Define the instance as uniquely addressable by calling endpointConfiguration.MakeUniquelyAddressable(discriminator);");
+                }
+
+                var localAddress = context.Settings.LocalAddress();
```

Comment:
yes but only for the subscriberAddress, before these changes, the replyto address was also pointing at the distributor?